frank/utilities.py

```diff
@@ -231,58 +324,90 @@ def cut_data_by_baseline(u, v, vis, weights, cut_range):
+        average of the variance of the real and imaginary components, as in
+        CASA's statwt. Otherwise the variance of the real part is used.
     nbins : int, default=300
-        Number of bins used to construct the histograms
+        Number of bins used.
+    log : bool, default=True
+        If True, the uv bins will be constructed in log space, otherwise linear
+        spaced bins will be used.
+    use_median : bool, default=False
+        If True all of the weights will be set to the median of the variance
+        estimated across the bins. Otherwise, the baseline dependent variance
+        will be used.
 
     Returns
     -------
-    wcorr_estimate : float
-        Correction factor by which to adjust the weights
-
-    weights_corrected : array, unit = Jy^-2
-        Corrected weights assigned to observed visibilities, of the form
-        :math:`1 / \sigma^2`
+    weights : array,
+        Esimtate weight for each uv point.
+
+    Notes
+    -----
+        - This function does not support datasets where the weights may be
+          expected to be different for different baselines, e.g. if the data
+          includes visibilities from telescopes of different sizes
+        - Bins with only one uv point will not have a variance estimate. The
+          median across all baselines is used instead.
     """
-
-    logging.info('  Estimating, applying correction factor to visibility weights')
-
-    baselines = np.hypot(u, v)
-    mu, edges = np.histogram(np.log10(baselines), weights=ReV, bins=nbins)
-    mu2, edges = np.histogram(np.log10(baselines), weights=ReV ** 2, bins=nbins)
-    N, edges = np.histogram(np.log10(baselines), bins=nbins)
-
-    centres = 0.5 * (edges[1:] + edges[:-1])
-
-    mu /= np.maximum(N, 1)
-    mu2 /= np.maximum(N, 1)
-
-    sigma = (mu2 - mu ** 2) ** 0.5
-    wcorr_estimate = sigma[np.where(sigma > 0)].mean()
-
-    weights_corrected = weights / wcorr_estimate ** 2
-
-    return wcorr_estimate, weights_corrected
-
+    logging.info('  Estimating visibility weights.')
+
+    q = np.hypot(u,v)
+    if log:
+        q = np.log(q)
+        q -= q.min()
+        
+    bin_width = (q.max()-q.min()) / nbins
+
+    uvBin = UVDataBinner(q, V, np.ones_like(q), bin_width)
+
+    if uvBin.bin_counts.max() == 1:
+        raise ValueError("No bin contains more than one uv point, can't "
+                         " estimate the variance. Use fewer bins.")
+
+    if np.iscomplex(V.dtype):
+        var = 0.5*(uvBin.error.real**2 + uvBin.error.imag**2) * uvBin.bin_counts
+    else:
+        var = uvBin.error.real**2 * uvBin.bin_counts
+
+    if use_median:
+        return np.full(np.len(u), np.median(var[uvBin.bin_counts > 1]))
+    else:
+        # For bins with 1 uv point, use the average of the adjacent bins
+        no_var = np.argwhere(uvBin.bin_counts == 1)
+        if len(no_var) > 0:
+            # Find the location of the bad points in the array of good points
```

Comment:
How are `good` vs `bad` points defined?
The definition should be in the code comments or, at least, in the function docstring (e.g. Notes section).

Reply:
To help reading the code, I'd suggest 
```suggestion
            # Find the location `loc` of the bad points in the array of good points
```